kobo/apps/hook/views/v2/hook_log.py

```diff
@@ -78,6 +96,43 @@ def get_queryset(self):
         # Django 1.9+, "select_related() prohibits non-relational fields for
         # nested relations."
         queryset = queryset.select_related('hook__asset')
+
+        # Filter on status
+        status = self.request.GET.get('status')
+        if status is not None:
+            if status not in map(str,VALID_STATUSES):
+                raise serializers.ValidationError(
+                    {'status': _('Value must be one of: ' +
+                                 ', '.join(map(str,VALID_STATUSES)))}
+                )
+            else:
+                queryset = queryset.filter(status=status)
+
+        # Filter on date range
+        start = self.request.GET.get('start')
+        if start is not None:
+          try:
+            start_date = datetime.fromisoformat(start)
+            if not start_date.tzname():
+                start_date=start_date.replace(tzinfo=timezone.utc)
+            queryset = queryset.filter(date_modified__gte=start_date)
+          except ValueError:
+            raise serializers.ValidationError(
+                {'start': _('Value must be a valid ISO-8601 date')}
+            )
+
+        end = self.request.GET.get('end')
+        if end is not None:
+          try:
+            end_date = datetime.fromisoformat(end)
+            if not end_date.tzname():
+                end_date=end_date.replace(tzinfo=timezone.utc)
+            queryset = queryset.filter(date_modified__lt=end_date)
+          except ValueError:
+            raise serializers.ValidationError(
+                {'end': _('Value must be a valid ISO-8601 date')}
+            )
```

Comment:
I'm not really up for it... I think I've already gone way beyond what can be reasonably expected of a casual community contributor. I've made it clear that I was a noob in Python/Django and yet was still encouraged to "give it a try", but then got sucked into having to learn a mostly undocumented set of standards (why isn't pep-8 compliance enforced by a linter instead of relying on manual reviews?), dev environment setup, test framework, that I think you will agree are a pretty steep learning curve.

If you hope to get more community contributions, I really think you need to invest into some dev documentation and toolchain setup...